### PR TITLE
Implement new H1 and H2 headings

### PR DESCRIPTION
**Pull request by bl-nero**
- Changes existing `h1` and `h2` typography classes.
- Introduces `H1` and `H2` wrappers for the `Text` component. The wrappers use actual HTML tags.
- Updates existing places where heading levels 1 and 2 should be used.
- Removes the horizontal lines from `FeatureHeaderTitle`
- Unifies typography of the "no data" states between unified resources and ordinary data tables

Contributes to https://github.com/gravitational/teleport/issues/44052
Must be merged together with https://github.com/gravitational/teleport.e/pull/4662.

All modified UIs verified either on storybook or in the actual app.

## Examples

Before:
Screenshot 2024-07-15 at 23 26 54

After:
Screenshot 2024-07-15 at 23 26 39

| Before | After |
|--------|------|
| Screenshot 2024-07-15 at 23 04 59 | Screenshot 2024-07-15 at 23 05 06 |
| Screenshot 2024-07-15 at 23 06 35 | Screenshot 2024-07-15 at 23 06 42 |

### CODE REVIEW

**Comment by bl-nero** [comment]
Note to reviewers: I have split actual code and snapshot updates into two separate commits for your convenience.

File: web/packages/teleport/src/components/Layout/Layout.tsx [hunk 1]
```diff
@@ -17,15 +17,14 @@
...
 
 import styled from 'styled-components';
-import { Flex, Text } from 'design';
+import { Flex, H1 } from 'design';
 import { alignItems, space } from 'design/system';
 
 /**
  * Header
  */
 const FeatureHeader = styled(Flex)`
```
Comment (gzdunek) [inline web/packages/teleport/src/components/Layout/Layout.tsx:26]: I'm not sure if this should be addressed in this PR, but spacing around `FeatureHeader` doesn't look good.
Now:
image

With `margin-top` and `margin-bottom` = 12px:
image

**Review by gzdunek — APPROVED** [review]
I tested Connect, didn't see anything unusual.

File: web/packages/teleport/src/components/Layout/Layout.tsx (hunk 1 above)
Comment (bl-nero) [reply web/packages/teleport/src/components/Layout/Layout.tsx:26]: Indeed, it looks better this way, but outside of the scope.

**Review by avatus — APPROVED** [review]
I poked around web. Seems fine

**Pull request merged**